instance/models/mixins/rabbitmq.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app model mixins - RabbitMQ
+"""
+
+# Imports #####################################################################
+
+import pathlib
+
+from django.conf import settings
+from django.db import models
+
+from instance.ansible import capture_playbook_output
+from instance.models.rabbitmq import RabbitMQUser
+
+
+# Classes #####################################################################
+
+class ReconfigurationFailed(Exception):
+    """Exception indicating that reconfiguring RabbitMQ failed."""
+
+
+class RabbitMQInstanceMixin(models.Model):
+    """
+    An instance that uses a RabbitMQ vhost with a set of users.
+    """
+    rabbitmq_vhost = models.CharField(max_length=16, blank=True)
+    rabbitmq_provider_user = models.ForeignKey(RabbitMQUser, related_name='provider_instance',
+                                               on_delete=models.CASCADE, blank=True, null=True)
+    rabbitmq_consumer_user = models.ForeignKey(RabbitMQUser, related_name='consumer_instance',
+                                               on_delete=models.CASCADE, blank=True, null=True)
+    rabbitmq_provisioned = models.BooleanField(default=False)
+
+    class Meta:
+        abstract = True
+
+    def _get_rabbitmq_vars(self, remove):
```

Comment:
Yeah I thought I might get a comment on this bit. ;) I'm inclined to agree. I'll opt for rendering the vars in a template - this will make it more legible and more in line with Ansible vars elsewhere in the codebase.